clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -3664,12 +3680,18 @@ static void RenderSSPOptions(const Driver &D, const ToolChain &TC,
           << A->getOption().getName() << Value;
       return;
     }
+    if (EffectiveTriple.isRISCV() && (Offset <= -2048 || Offset >= 2048)) {
```

Comment:
I don't know a lot about stack protector. Where does this range come from? Is this trying to be 12-bit signed immediate? If so it should exclude 2048.

Reply:
Should probably have a comment and use IsInt<12>() instead of range checks.

Reply:
Yeah, this is used as an offset from the TLS base register (tp). GCC doesn't allow arbitrary offsets for this value, permitting only a value that can be encoded as an immediate value in the load instruction. I mirrored those limits (slightly incorrectly) in my patch. Given that llvm can support arbitrary values as the compiler is happy to use multiple instructions as required, I think it's probably best to just remove this test instead of introducing an artificial limit to match gcc's behavior?

Reply:
Ok. If the backend supports any value, we can drop this.

Reply:
Done, and new bits pushed.